x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/validations.ts

```diff
@@ -59,37 +66,37 @@ export const ROLLOVER_EMPTY_VALIDATION = 'ROLLOVER_EMPTY_VALIDATION';
  */
 export const rolloverThresholdsValidator: ValidationFunc = ({ form, path }) => {
   const fields = form.getFields();
-  if (
-    !(
-      fields[ROLLOVER_FORM_PATHS.maxAge]?.value ||
-      fields[ROLLOVER_FORM_PATHS.maxDocs]?.value ||
-      fields[ROLLOVER_FORM_PATHS.maxSize]?.value
-    )
-  ) {
-    if (path === ROLLOVER_FORM_PATHS.maxAge) {
-      return {
-        code: ROLLOVER_EMPTY_VALIDATION,
-        message: i18nTexts.editPolicy.errors.maximumAgeRequiredMessage,
-      };
-    } else if (path === ROLLOVER_FORM_PATHS.maxDocs) {
-      return {
-        code: ROLLOVER_EMPTY_VALIDATION,
-        message: i18nTexts.editPolicy.errors.maximumDocumentsRequiredMessage,
-      };
-    } else {
-      return {
-        code: ROLLOVER_EMPTY_VALIDATION,
-        message: i18nTexts.editPolicy.errors.maximumSizeRequiredMessage,
-      };
+  // At least one rollover field needs a value specified for this action.
+  const someRolloverFieldHasAValue = rolloverFieldPaths.some((rolloverFieldPath) =>
+    Boolean(fields[rolloverFieldPath]?.value)
+  );
+  if (!someRolloverFieldHasAValue) {
+    const errorToReturn: ValidationError = {
+      code: ROLLOVER_VALUE_REQUIRED_VALIDATION_CODE,
+      message: '', // We need to map the current path to the corresponding validation error message
+    };
+    switch (path) {
+      case ROLLOVER_FORM_PATHS.maxAge:
+        errorToReturn.message = i18nTexts.editPolicy.errors.maximumAgeRequiredMessage;
+        break;
+      case ROLLOVER_FORM_PATHS.maxDocs:
+        errorToReturn.message = i18nTexts.editPolicy.errors.maximumDocumentsRequiredMessage;
+        break;
+      case ROLLOVER_FORM_PATHS.maxPrimaryShardSize:
+        errorToReturn.message = i18nTexts.editPolicy.errors.maximumPrimaryShardSizeRequiredMessage;
+        break;
+      default:
+        errorToReturn.message = i18nTexts.editPolicy.errors.maximumSizeRequiredMessage;
     }
+    return errorToReturn;
   } else {
-    fields[ROLLOVER_FORM_PATHS.maxAge].clearErrors(ROLLOVER_EMPTY_VALIDATION);
-    fields[ROLLOVER_FORM_PATHS.maxDocs].clearErrors(ROLLOVER_EMPTY_VALIDATION);
-    fields[ROLLOVER_FORM_PATHS.maxSize].clearErrors(ROLLOVER_EMPTY_VALIDATION);
+    rolloverFieldPaths.forEach((rolloverFieldPath) => {
+      fields[rolloverFieldPath]?.clearErrors(ROLLOVER_VALUE_REQUIRED_VALIDATION_CODE);
+    });
   }
 };
 
-export const integerValidator: ValidationFunc<FormInternal, string, string> = (arg) => {
+export const integerValidator: ValidationFunc<any, string, string> = (arg) => {
```

Comment:
I see why you've added the `any` here. I solved this kind of problem by _not_ typing the schema

```js
// schema.ts (L136)
export const getSchema = (isCloudEnabled: boolean): FormSchema => ({
```

So I could correctly type here the validation.